docs/src/main/tut/enum.md

```diff
@@ -0,0 +1,24 @@
+---
+layout: default
+title:  "Enum"
+source: "core/src/main/scala/Enum.scala"
+---
+# Enum
+
+`Enum` represents discrete operations that can be performed on a type `A`
+
+These operations are presented by the following functions.
+
+- `def succ(x)`:		Returns the successor value of `x`
+- `def pred(x)`: 	Returns the predecessor value of `x`
+- `def adj(i, j)`:	Returns if `i` and `j` are consecutives ( `succ(i) is j`)
+
+An example of the discrete operation on interger values could be: 
+
+```
```

Comment:
look at the other sources in this directory, and take a look at the docs for tut. the idea is that if you label these sections with tut, they will be compiled, and you see the output of a repl session in the resulting .md file. Compare the list.md to the resulting html page.

This way we make sure the code in our docs is always up to date